### DISCUSSION TITLE
How can I acces to .env variables in a JS File using Nuxt?

### DISCUSSION THREAD

**Discussion by raul-espinosa-go** [nuxt/nuxt#27054] [Questions]
"Hello. I have a JS file within a Nuxt3 project where I handle requests and fetches, and I would like to import some environment variables. The issue is that since it's not a .vue file, I can't access it with the runtime, and being SSR, I can't access it with the dotenv library either. What solution do you suggest? Thank you very much."

**Comment by TheAlexLichter** [comment]
You can pass in the necessary values when you call the functions in your app. I've given [an example here](https://www.youtube.com/watch?v=jXH8Tr-exhI) on how to use a custom fetcher + providing values from the runtimeConfig to it

**Reply by raul-espinosa-go** [reply]
I will try it, thanks!

What I was trying to do is to save the URL in my .env for accessing a socket server in a JS file within the Nuxt project like this:

### JS File
```
const config = useRuntimeConfig();
const environment = config.public;

const url = environment.SOCKET_URI;

export const socket = io(url);
```

### .env
```
SOCKET_URI=http://localhost:8080
```

### nuxt.config.ts
```
  runtimeConfig: {
    public: {
      SOCKET_URI: process.env.SOCKET_URI,
    },
```
And when I upload the project to the server, after building/generating it and everything else, I encounter an error of [nuxt] instance unavailable. So, I understand that the issue is the one I mentioned earlier.
